File: smartclass-frontend-main/src/api/aiAvatar.ts
import type { AxiosResponse } from 'axios';
import apiClient from './index';
import { buildApiUrl } from '@/utils/api';

interface BaseResponse<T> {
  code?: number;
  data?: T;
  message?: string;
}

interface PageResponse<T> {
  records?: T[];
  total?: number;
  current?: number;
  size?: number;
  pages?: number;
}

export interface AiAvatarInfo {
  id?: number;
  name?: string;
  baseUrl?: string;
  description?: string;
  avatarImgUrl?: string;
  tags?: string;
  personality?: string;
  abilities?: string;
  isPublic?: number;
  status?: number;
  usageCount?: number;
  rating?: number;
  ratingCount?: number;
  creatorId?: number;
  sort?: number;
  createTime?: string;
  updateTime?: string;
}

export interface AiAvatarPageQuery {
  current?: number;
  pageSize?: number;
  keyword?: string;
}

const unwrap = <T>(response: AxiosResponse<BaseResponse<T>>, fallbackMessage: string): T => {
  const payload = response.data;
  if (payload?.code === 0 && payload.data !== undefined) {
    return payload.data;
  }

  throw new Error(payload?.message || fallbackMessage);
};

export const queryAiAvatarPage = async (
  query: AiAvatarPageQuery = {},
): Promise<PageResponse<AiAvatarInfo>> => {
  const params: Record<string, string | number> = {
    current: query.current ?? 1,
    pageSize: query.pageSize ?? 20,
  };

  const keyword = query.keyword?.trim();
  if (keyword) {
    params.keyword = keyword;
  }

  const response = await apiClient.get<BaseResponse<PageResponse<AiAvatarInfo>>>(
    '/api/ai-avatars/page',
    { params },
  );

  return unwrap(response, '智慧体列表获取失败');
};

export const queryAiAvatarDetail = async (id: number): Promise<AiAvatarInfo> => {
  const response = await apiClient.get<BaseResponse<AiAvatarInfo>>(`/api/ai-avatars/${id}`);
  return unwrap(response, '智慧体详情获取失败');
};

export const resolveAiAvatarImage = (avatarUrl?: string): string => {
  const value = avatarUrl?.trim();
  if (!value) {
    return '';
  }

  if (
    value.startsWith('http://') ||
    value.startsWith('https://') ||
    value.startsWith('//') ||
    value.startsWith('data:') ||
    value.startsWith('blob:')
  ) {
    return value;
  }

  return buildApiUrl(value.startsWith('/') ? value : `/${value}`);
};

File: learning-system/src/main/java/com/ttbt/smartclass/model/entity/AiAvatar.java
package com.ttbt.smartclass.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

/**
 * AI 数字人实体。
 */
@TableName(value = "ai_avatar")
@Data
public class AiAvatar implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    @TableField("base_url")
    private String baseUrl;

    private String description;

    @TableField("avatar_img_url")
    private String avatarImgUrl;

    @TableField("avatar_auth")
    private String avatarAuth;

    private String tags;

    private String personality;

    private String abilities;

    @TableField("is_public")
    private Integer isPublic;

    private Integer status;

    @TableField("usage_count")
    private Integer usageCount;

    private BigDecimal rating;

    @TableField("rating_count")
    private Integer ratingCount;

    @TableField("creator_id")
    private Long creatorId;

    private Integer sort;

    @TableField("create_time")
    private Date createTime;

    @TableField("update_time")
    private Date updateTime;

    @TableField("is_delete")
    private Integer isDelete;

    @TableField(exist = false)
    private static final long serialVersionUID = 1L;
}
